### PR TITLE
Added a target to verify nupkgs with NugetPackageVerifier

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Naming suggestions welcome :smiley: 
@pranavkm @Eilon

### CODE REVIEW

File: build/_k-standard-goals.shade [hunk 1]
```diff
@@ -187,6 +189,18 @@ default Quiet='${ false }'
     }
   }
 
+#nuget-verify target='test' if='File.Exists(PACKAGELIST_JSON_FILENAME)' description='Verify if all the packages are generated properly'
+  exec program='cmd' commandline='/C dnu commands install --source ${DNX_TOOLS_FEED} NugetPackageVerifier --packages ${BASE_DIR}\commands\' if='ShouldVerifyNupkgs'
```
Comment (Eilon) [inline build/_k-standard-goals.shade]: `NugetPackageVerifier` --> `NuGetPackageVerifier` (capital "G" in NuGet)

**Comment by Eilon** [comment]
:shipit:

File: build/_k-standard-goals.shade [hunk 2]
```diff
@@ -13,6 +13,8 @@ default BUILD_DIR='${Path.Combine(TARGET_DIR, "build")}'
 default TEST_DIR='${Path.Combine(TARGET_DIR, "test")}'
 default Configuration='${E("Configuration")}'
 default Quiet='${ false }'
+default PACKAGELIST_JSON_FILENAME = 'packages.json'
+default DNX_TOOLS_FEED = 'http://www.myget.org/F/dnxtools/api/v2'
```
Comment (pranavkm) [inline build/_k-standard-goals.shade]: `https`

File: build/_k-standard-goals.shade [hunk 3]
```diff
@@ -187,6 +189,18 @@ default Quiet='${ false }'
     }
   }
 
+#nuget-verify target='test' if='File.Exists(PACKAGELIST_JSON_FILENAME)' description='Verify if all the packages are generated properly'
```
Comment (pranavkm) [inline build/_k-standard-goals.shade]: Could we run this target conditionally? I don't think there's a great deal of utility in running this on local builds all the time.

File: build/_k-standard-goals.shade (hunk 1 above)
Comment (pranavkm) [inline build/_k-standard-goals.shade]: `${BASE_DIR}\commands\' => `Path.Combine`. Also does Sake let you split lines? Kinda hard to read long scrolling lines.

File: build/_k-standard-goals.shade [hunk 4]
```diff
@@ -187,6 +189,18 @@ default Quiet='${ false }'
...
+  exec program='cmd' commandline='/C dnu commands install --source ${DNX_TOOLS_FEED} NugetPackageVerifier --packages ${BASE_DIR}\commands\' if='ShouldVerifyNupkgs'
+  exec program='cmd' commandline='/C ${BASE_DIR}/commands/nugetverify ${BUILD_DIR} ${BASE_DIR}\${PACKAGELIST_JSON_FILENAME}' if='ShouldVerifyNupkgs'
```
Comment (pranavkm) [inline build/_k-standard-goals.shade]: Also Path.Combine.

File: build/_k-standard-goals.shade [hunk 5]
```diff
@@ -227,6 +241,11 @@ default Quiet='${ false }'
     E("KOREBUILD_PARALLEL", "1");
    }
 
+#--verifynupkgs
```
Comment (pranavkm) [inline build/_k-standard-goals.shade]: Let's just have the environment variable. We'll figure out a way to make it work via passed in arguments once @victorhurdugaci updates sake.

**Comment by pranavkm** [comment]
:watch:

File: build/_k-standard-goals.shade (hunk 3 above)
Comment (Eilon) [inline build/_k-standard-goals.shade]: Isn't it already conditional?

Comment (Eilon) [inline build/_k-standard-goals.shade]: The stuff inside this has `if='ShouldVerifyNupkgs'`

File: build/_k-standard-goals.shade [hunk 6]
```diff
@@ -13,6 +13,8 @@ default BUILD_DIR='${Path.Combine(TARGET_DIR, "build")}'
 default TEST_DIR='${Path.Combine(TARGET_DIR, "test")}'
 default Configuration='${E("Configuration")}'
 default Quiet='${ false }'
+default PACKAGELIST_JSON_FILENAME = 'packages.json'
```
Comment (victorhurdugaci) [inline build/_k-standard-goals.shade]: where is this file? Who's generating it?

File: build/_k-standard-goals.shade (hunk 1 above)
Comment (victorhurdugaci) [inline build/_k-standard-goals.shade]: I'm a little concerned about this. The tool will run using the active DNX. We don't have a way of "pinning" it to a particular DNX. There is a chance that future DNX changes will break it.

File: build/_k-standard-goals.shade (hunk 6 above)
Comment (Eilon) [inline build/_k-standard-goals.shade]: It will be checked into certain repos.

File: build/_k-standard-goals.shade (hunk 1 above)
Comment (Eilon) [inline build/_k-standard-goals.shade]: @victorhurdugaci true, and maybe eventually it'll use a private DNX, but in the meantime we'll just update the tool if it breaks.

File: build/_k-standard-goals.shade [hunk 7]
```diff
@@ -293,6 +312,11 @@ functions @{
     {
         get { return !string.IsNullOrEmpty(E("KOREBUILD_PARALLEL")); }
     }
+
+    bool ShouldVerifyNupkgs
+    {
+      get { return !string.IsNullOrEmpty(E("VERIFY_NUPKGS")); }
```
Comment (victorhurdugaci) [inline build/_k-standard-goals.shade]: shouldn't you check against value "1"? What if I set VERIFY_NUPKS to 0?

File: build/_k-standard-goals.shade [hunk 8]
```diff
@@ -227,6 +241,11 @@ default Quiet='${ false }'
     E("KOREBUILD_PARALLEL", "1");
    }
 
+#--verifynupkgs
+   @{
+    E("VERIFY_NUPKGS", "1");
```
Comment (victorhurdugaci) [inline build/_k-standard-goals.shade]: `KOREBUILD_VERIFY_NUKPS`

File: build/_k-standard-goals.shade (hunk 1 above)
Comment (victorhurdugaci) [inline build/_k-standard-goals.shade]: btw, where's the code for the verifier?

Comment (Eilon) [inline build/_k-standard-goals.shade]: https://github.com/aspnet/DnxTools/tree/master/NuGetPackageVerifier

File: build/_k-standard-goals.shade [hunk 9]
```diff
@@ -187,6 +189,18 @@ default Quiet='${ false }'
     }
   }
 
+#nuget-verify target='test' if='File.Exists(PACKAGELIST_JSON_FILENAME)' description='Verify if all the packages are generated properly'
+  exec program='cmd' commandline='/C dnu commands install --source ${DNX_TOOLS_FEED} NugetPackageVerifier --packages ${BASE_DIR}\commands\' if='ShouldVerifyNupkgs'
+  exec program='cmd' commandline='/C ${BASE_DIR}/commands/nugetverify ${BUILD_DIR} ${BASE_DIR}\${PACKAGELIST_JSON_FILENAME}' if='ShouldVerifyNupkgs'
+  @{
+    var directory = Path.Combine(BASE_DIR, "commands");
```
Comment (victorhurdugaci) [inline build/_k-standard-goals.shade]: "commands" -> DRY. use a const

Comment (Eilon) [inline build/_k-standard-goals.shade]: It isn't *really* repeated, it's just the same value. There's nothing to "refactor" here. This code is perfectly readable and hiding things behind consts will harm that...

File: build/_k-standard-goals.shade (hunk 3 above)
Comment (pranavkm) [inline build/_k-standard-goals.shade]: Specifying it this way means that the target doesn't get defined at all. In which case, you don't have to have this check on every line

File: build/_k-standard-goals.shade (hunk 1 above)
Comment (ajaybhargavb) [inline build/_k-standard-goals.shade]: The `dev` branch has the latest code.

Comment (pranavkm) [inline build/_k-standard-goals.shade]: Isn't there a switch to include the DNX as part of the command? That should suffice, right?

Comment (Eilon) [inline build/_k-standard-goals.shade]: Oh yeah there is, but does it work for command line apps that are installed as commands? Maybe it does...

Comment (ajaybhargavb) [inline build/_k-standard-goals.shade]: I checked `dnu commands install --help` and I don't see a switch to include the DNX.

Comment (Eilon) [inline build/_k-standard-goals.shade]: @ajaybhargavb I'd log it in the DnxTools repo to track.

Comment (ajaybhargavb) [inline build/_k-standard-goals.shade]: https://github.com/aspnet/DnxTools/issues/2

Comment (pranavkm) [inline build/_k-standard-goals.shade]: Wouldn't it be in the `dnu pack` \ `dnu deploy` step?

**Comment by ajaybhargavb** [comment]
Updated.

**Comment by Eilon** [comment]
:shipit: :shipit: :smile:

File: build/_k-standard-goals.shade [hunk 10]
```diff
@@ -187,6 +189,18 @@ default Quiet='${ false }'
...
   }
 
+#nuget-verify target='test' if='File.Exists(PACKAGELIST_JSON_FILENAME) && ShouldVerifyNupkgs' description='Verify if all the packages are generated properly'
+  var commandsDirectory = '${Path.Combine(BASE_DIR, "commands")}'
+  exec program='cmd' commandline='/C dnu commands install --source ${DNX_TOOLS_FEED} NuGetPackageVerifier --packages "${commandsDirectory}"'
+  exec program='cmd' commandline='/C ${Path.Combine(commandsDirectory, "nugetverify")} "${BUILD_DIR}" "${Path.Combine(BASE_DIR, PACKAGELIST_JSON_FILENAME)}"'
+  @{
+    if (Directory.Exists(commandsDirectory))
+    {
+      var directoryInfo = new DirectoryInfo(commandsDirectory);
```
Comment (pranavkm) [inline build/_k-standard-goals.shade]: `Directory.Delete(commandsDirectory, recursive: true);`

**Comment by pranavkm** [comment]
:shipit:

**Pull request merged**